### PR TITLE
Remove over logging in the pool manager

### LINKED ISSUE #728: Handling of faults tends to lead to duplication with external handling and over-verbose logging
Exceptions thrown by various Java functions called rom HTTP4s are (at least sometimes) handled with full logging of the exception (including stack dumps) within HTTPS, but then still propagated out, where the condition needs to be also handled by the calling code.

This has two impacts:

1) Some logical duplication of effort between HTTP4S internal code, and external calling code
2) Inability to control logging behavior in a calling service/app

Suggestion is that internal (to HTTP4s) handling that propagates the fault to the caller (see example below) should not log (or at least not log verbosely with full stack information), since the caller anyway has to handle the fault.

Known example is when the `PoolManager` attempts to establish a connection to  server that is not contactable.  This results in a Java exception with the stack dump below, and then propagates the same exception in the resulting `Task[Response]` the client ultimately surfaces:

```
java.nio.channels.UnresolvedAddressException: null
           at sun.nio.ch.Net.checkAddress(Net.java:101) ~[na:1.8.0_102]
           at sun.nio.ch.UnixAsynchronousSocketChannelImpl.implConnect(UnixAsynchronousSocketChannelImpl.java:301) ~[na:1.8.0_102]
           at sun.nio.ch.AsynchronousSocketChannelImpl.connect(AsynchronousSocketChannelImpl.java:209) ~[na:1.8.0_102]
           at org.http4s.blaze.channel.nio2.ClientChannelFactory.connect(ClientChannelFactory.scala:27) ~[org.http4s.blaze-core_2.11-0.11.0.jar:0.11.0]
           at org.http4s.client.blaze.Http1Support.org$http4s$client$blaze$Http1Support$$buildPipeline(Http1Support.scala:72) ~[org.http4s.http4s-blaze-client_2.11-0.12.4.jar:0.12.4]
           at org.http4s.client.blaze.Http1Support$$anonfun$makeClient$1.apply(Http1Support.scala:67) ~[org.http4s.http4s-blaze-client_2.11-0.12.4.jar:0.12.4]
           at org.http4s.client.blaze.Http1Support$$anonfun$makeClient$1.apply(Http1Support.scala:67) ~[org.http4s.http4s-blaze-client_2.11-0.12.4.jar:0.12.4]
           at org.http4s.util.TaskFunctions$$anonfun$futureToTask$1.apply(Task.scala:21) ~[org.http4s.http4s-core_2.11-0.12.4.jar:0.12.4]
           at org.http4s.util.TaskFunctions$$anonfun$futureToTask$1.apply(Task.scala:20) ~[org.http4s.http4s-core_2.11-0.12.4.jar:0.12.4]
```

### PR DESCRIPTION
**Pull request by ioganegambaputifonguser**
fixes #728 Remove over logging in the pool manager. Add log4s configuration for the blaze-client test.

### CODE REVIEW

File: blaze-client/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n${LOGBACK_EXCEPTION_PATTERN:-%throwable}</pattern>
+        </encoder>
+    </appender>
+
+    <root level="${LOGBACK_ROOT_LEVEL:-WARN}">
+        <appender-ref ref="STDOUT" />
+    </root>
+</configuration>
```
Comment (rossabaker) [inline blaze-client/src/test/resources/logback-test.xml]: Surprise this file is necessary.  This has a test->test dependency from testing, so why is this not on the classpath already?

Comment (ChristopherDavenport) [reply blaze-client/src/test/resources/logback-test.xml]: Right, I'd feel much more comfortable without this file.

**Comment by ioganegambaputifonguser** [comment]
Deleted

**Comment by rossabaker** [comment]
Cherry-picked and merged to master.  Thanks!

**Pull request closed without merging**